sdk/storage/Azure.Storage.Blobs/src/Models/BlobInfo.cs

```diff
@@ -369,4 +410,304 @@ public static partial class BlobsModelFactory
+        /// </summary>
+        public string VersionId { get; internal set; }
+
+        /// <summary>
+        /// The value of this header indicates whether version of this blob is a current version, see also x-ms-version-id header.
+        /// </summary>
+        public bool IsCurrentVersion { get; internal set; }
+
+        /// <summary>
+        /// The number of tags associated with the blob
+        /// </summary>
+        public long TagCount { get; internal set; }
+
+        /// <summary>
+        /// The time this blob will expire.
+        /// </summary>
+        public DateTimeOffset ExpiresOn { get; internal set; }
+
+        /// <summary>
+        /// If this blob has been sealed
+        /// </summary>
+        public bool IsSealed { get; internal set; }
+
+        /// <summary>
+        /// Internal. Parses Object Replication Policy ID from Rule ID and sets the Policy ID.
+        /// </summary>
+        /// <returns></returns>
+        internal IDictionary<string, IDictionary<string, string>> ParseObjectReplicationIds()
+        {
+            if (_properties.ObjectReplicationRules.Count == 0 ||
+                (_properties.ObjectReplicationRules.Count > 0 &&
+                (_properties.ObjectReplicationRules.First().Key == "policy-id")))
+            {
+                return default;
+            }
+            IDictionary<string, IDictionary<string, string>> OrProperties = new Dictionary<string, IDictionary<string, string>>();
+            foreach (KeyValuePair<string, string> status in _properties.ObjectReplicationRules)
+            {
+                string[] ParsedIds = status.Key.Split('_');
+                if (OrProperties.ContainsKey(ParsedIds[0]))
+                {
+                    OrProperties[ParsedIds[0]].Add(ParsedIds[1], status.Value);
+                }
+                else
+                {
+                    IDictionary<string, string> NewRuleStatus = new Dictionary<string, string>();
+                    NewRuleStatus.Add(ParsedIds[1], status.Value);
+                    OrProperties.Add(ParsedIds[0], NewRuleStatus);
+                }
+            }
+            return OrProperties;
+        }
+
+        /// <summary>
+        /// Creates a new BlobPropertiesInternal instance
+        /// </summary>
+        public BlobProperties()
+        {
+            Metadata = new Dictionary<string, string>(System.StringComparer.OrdinalIgnoreCase);
+            ObjectReplicationSourceProperties = ParseObjectReplicationIds();
+        }
+
+        /// <summary>
+        /// Creates a new BlobProperties object backed by BlobPropertiesInternal
+        /// </summary>
+        /// <param name="properties">The BlobPropertiesInternal returned with the request</param>
+        public BlobProperties(BlobPropertiesInternal properties)
```

Comment:
Maybe make this an extension method on `BlobPropertiesInteral`?  Either way is fine, but most of the model conversions in Data Lake use extension methods.